logstash-core/lib/logstash/settings.rb

```diff
@@ -140,6 +137,11 @@ def validate_all
       end
     end
 
+    def ==(other)
+      return false unless other.kind_of?(::LogStash::Settings)
+      self.to_hash == other.to_hash
+    end
```

Comment:
I'm not sure of your concern here, I added a few tests to the ArrayCoercible in https://github.com/elastic/logstash/pull/6525/files#diff-bb0f568c71d6288772b068b4f1830c33
let me know if this helps